explore/src/main/scala/explore/constraints/TimingWindowsPanel.scala

```diff
@@ -208,11 +244,29 @@ object TimingWindowsPanel:
               case None     => Callback.empty
             }
 
-        val selectedStartsOn      = selectedTW.zoom(TimingWindow.startsOn)
-        val selectedCloseOn       = selectedTW.zoom(TimingWindow.closeOn)
-        val selectedRemainOpenFor = selectedTW.zoom(TimingWindow.remainOpenFor)
-        val selectedRepeatPeriod  = selectedTW.zoom(TimingWindow.repeatPeriod)
-        val selectedRepeatNTimes  = selectedTW.zoom(TimingWindow.repeatFrequency)
+        val selectedType         = selectedTW.zoom(TimingWindow.windowType)
+        val selectedfrom         = selectedTW.zoom(TimingWindow.from)
+        val selectedThrough      = selectedTW.zoom(TimingWindow.through)
+        val selectedFiniteSpan   = selectedTW.zoom(TimingWindow.finiteSpan)
+        val selectedRepeatPeriod = selectedTW.zoom(TimingWindow.repeatPeriod)
+        val selectedRepeatNTimes = selectedTW.zoom(TimingWindow.repeatFrequency)
+
+        // TODO Should we move this to lucuma-ui?
+        val HMPartialRegEx                  = "\\d*(:\\d{0,2})?".r
+        val hmChangeAuditor: ChangeAuditor  = ChangeAuditor.accept.allow(HMPartialRegEx.matches)
+        val HMSPartialRegEx                 = "\\d*(:\\d{0,2}(:\\d{0,2})?)?".r
```

Comment:
I suppose the hour part could also be `\\d{0,2}`?

Reply:
It was intentionally left unbounded since this is used to represent a time span. This way you can define a window open for, say, 10 days.